server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -16,183 +16,185 @@ def get(self, section: str, option: str):
 
 
 class TestProcessTb:
-    @pytest.fixture(autouse=True)
-    def logger(self, test_logger):
-        # Setup the configuration and logger
-        self.logger = test_logger
-
-    def test_get_receive_dir_failed(self, monkeypatch):
+    def test_get_receive_dir_failed(self, monkeypatch, test_logger):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return ""
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(ValueError) as e:
-            ProcessTb(MockConfig, self.logger)
+            ProcessTb(MockConfig, test_logger)
 
         assert expected_error_msg in str(e)
 
-    def test_get_receive_dir_value_failed(self, monkeypatch):
+    def test_get_receive_dir_value_failed(self, monkeypatch, test_logger):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
-                return receive_dir / "wrong-directory"
-            assert False
+                return fixture_path / "wrong-directory"
+            assert (
+                False
+            ), f"Unexpected value for option, {section!r}/{option!r}, in mocked config.get()"
 
-        expected_error_path = str(receive_dir / "wrong-directory-002")
+        expected_error_path = str(fixture_path / "wrong-directory-002")
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(Exception) as e:
-            ProcessTb(MockConfig, self.logger)
+            ProcessTb(MockConfig, test_logger)
 
         assert expected_error_path in str(e)
 
-    def test_process_tb_file_not_found_error(self, monkeypatch, caplog):
+    def test_process_tb_file_not_found_error(self, monkeypatch, caplog, test_logger):
         """checks normal processing when tar ball is not present"""
         tb_path = fixture_path / "bad_tarball"
 
-        def mock_glob(*args, **kwargs):
-            tb_path = fixture_path / "bad_tarball-002" / "bad_log.tar.xz.md5"
-            return [tb_path]
-
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
-            if str(tb).startswith(str(tb_path)):
+            bad_tb = self.receive_dir / bad_tarball
+            if bad_tb == tb:
                 raise FileNotFoundError(f"No such file or directory: '{tb.name}'")
```

Comment:
How does this work without a mock for `glob()`?  Does `tb_path / "bad_tarball" / bad_tarball` actually exist?

(The same question applies to the other test scenarios where you've removed the `mock_glob()` definitions....)

Reply:
yes, they exist in fixtures folder